braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java

```diff
@@ -397,12 +437,85 @@ public Braze getUnderlyingInstance() {
         return braze;
     }
 
+    //////////////////////////////////////////////////
+    // UTIL Methods
+    //////////////////////////////////////////////////
     private static Date dateFromString(String date) {
         SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd", Locale.US);
         try {
             return formatter.parse(date);
         } catch (Exception e) {
+            RudderLogger.logError(e);
             return null;
         }
     }
+
+    // compare two address objects and return accordingly
+    private static boolean compareAddress(@Nullable RudderTraits.Address curr, @Nullable RudderTraits.Address prev) {
+        if (prev == null && curr != null) {
+            return true;
+        }
+
+        return prev != null && curr != null
+                && prev.getCity().equals(curr.getCity())
+                && prev.getCountry().equals(curr.getCountry())
+                && prev.getPostalCode().equals(curr.getPostalCode())
+                && prev.getState().equals(curr.getState())
+                && prev.getStreet().equals(curr.getStreet());
```

Comment:
```suggestion
        return prev != null && curr != null
                && prev.getCity().equals(curr.getCity())
                && prev.getCountry().equals(curr.getCountry());
```

Can we remove the check for `PostalCode`, `State` and `Street`, since over [here](https://github.com/rudderlabs/rudder-integration-braze-android/blob/b615d53eebe8bf2398e0905f81320e2ba158908b/braze/src/main/java/com/rudderstack/android/integration/braze/BrazeIntegrationFactory.java#L344), we are only concerned about `City` and `Country`.

Reply:
This logic needs to be improved further, as it will throw runtime exception in this scenario:

- Make first identify call with `City` and `Country` value.
- Make second identify call with `City` value only.

Now, this line will throw runtime error in the second identify call:
> && prev.getCountry().equals(curr.getCountry());

as `prev.getCountry()` returns null in second case and we are using `equals` operation on the `null` value.